### PR TITLE
planner: update the underlying binding cache structure

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #51347

Problem Summary: planner: update the underlying binding cache structure

### What changed and how does it work?

Just refactor to replace the underlying cache structure to use `ristretto.Cache`, which is faster, thread-safe and supports cost-based eviction and can help further simplify our code.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by time-and-fate — APPROVED** [review]

File: pkg/bindinfo/internal/testutil.go [hunk 1]
```diff
@@ -26,9 +26,11 @@ import (
 )
 
 // UtilCleanBindingEnv cleans the binding environment.
-func UtilCleanBindingEnv(tk *testkit.TestKit, dom *domain.Domain) {
+func UtilCleanBindingEnv(tk *testkit.TestKit, _ *domain.Domain) {
```
Comment (hawkingrei) [inline pkg/bindinfo/internal/testutil.go:29]:
```suggestion
func UtilCleanBindingEnv(tk *testkit.TestKit) {
```

Comment (qw4990) [reply pkg/bindinfo/internal/testutil.go]: done

**Review by hawkingrei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**